src/object_cmd.cpp

```diff
@@ -121,7 +122,9 @@ void BuildObject(ObjectType type, TileIndex tile, CompanyID owner, Town *town, u
 			Company::Get(owner)->infrastructure.water++;
 			DirtyCompanyInfrastructureWindows(owner);
 		}
+		bool remove = IsDockingTile(t);
 		MakeObject(t, owner, o->index, wc, Random());
+		if (remove) RemoveDockingTile(t);
```

Comment:
I honestly don't know the answer, but why does this need a variable? Can't we just do `RemoveDockingTile` before `MakeObject`?

Reply:
No. RemoveDockingTile examines the state of the tile.

Reply:
So `RemoveDockingTileIfIFeelLikeIt`, gotcha. Tnx :)